app/models/partnership.rb

```diff
@@ -67,4 +68,10 @@ def cohort_challenge_deadline
   def update_analytics
     Analytics::UpsertECFPartnershipJob.perform_later(partnership: self) if saved_changes?
   end
+
+  def set_challenge_deadline_by_cohort
+    if cohort.start_year == 2023 && created_at < Date.new(2023, 10, 17)
```

Comment:
I think what we need here is something like:

```
CHALLENGE_PERIOD_AFTER_CREATION = 2.weeks

def set_challenge_deadline
  [
     cohort.academic_year_start_date + CHALLENGE_PERIOD_SINCE_ACADEMIC_YEAR_START,
     Time.current + CHALLENGE_PERIOD_AFTER_CREATION
  ].max
end
```
so that the deadline will be 31 Oct each academic year (1 Sept + 2.months) for partnerships reported before Oct, 17th
or 2.weeks after the creation for partnerships reported after Oct, 17th

Reply:
is two weeks the default already?
Furthermore the ask in the ticket is to explicitly do this for 2023 only.
Saying so, do you think we still need the above?

Reply:
Yes agree with @ltello's suggestion